daemon/src/model/cfd.rs

```diff
@@ -841,6 +909,169 @@ mod tests {
         let blocks = duration.as_blocks();
         assert!(blocks - error_margin < 0.1 && blocks + error_margin > 0.1);
     }
+
+    #[test]
+    fn test_profit_calculation_no_profit() {
```

Comment:
These tests are very repetitive. We could declare a single test function and run a list of values through it to make this less verbose.